Update hanyuu_commands.py

Annih7 opened the pull request:
for .queue length, let l work for length, ex: .q l should work now
same for .fave last, should allow .fave l
    also allow .favorite instead of .fave
did last>l for .unfave, mostly for consistency's sake

for .random fave, let "favorite" and "f" work in place of "fave"
oh god, do i have to put this in 2 more times for .random fave (name)? I think I did it right....

.lucky can be .l

what the hell, make .fl(ist) be .f(ave|avorite)l(ist)
    does anyone even use this? (nope)

Annih7 committed:
Update hanyuu_commands.py

for .queue length, let l work for length, ex: .q l should work now
same for .fave last, should allow .fave l
    also allow .favorite instead of .fave, not that anyone will use it
did last>l for .unfave, mostly for consistency's sake

for .random fave, let "favorite" and "f" work in place of "fave"
oh god, do i have to put this in 2 more times for .random fave (name)? I think I did it right....

.lucky can be .l

what the hell, make .fl(ist) be .f(ave|avorite)l(ist)
    does anyone even use this? (nope)

Wessie commented on hanyuu_commands.py:
```diff
@@ -362,12 +362,12 @@ def request_from_list(songs):
-    elif (re.match(r"^fave (.*)", command)):
-        fave_nick = re.match(r"^fave (.*)", command).groups()[1]
+    elif (re.match(r"^f(ave|avorite) (.*)", command)):
+        fave_nick = re.match(r"^f(ave|avorite) (.*)", command).groups()[1]
```
Should be `r"^f(ave|avorite)? (.*)"`

Wessie commented on hanyuu_commands.py:
```diff
@@ -393,7 +393,7 @@ def request_from_list(songs):
 
 def lucky(server, nick, channel, text, hostmask):
-    match = re.match(r"^(?P<mode>[.!@])lucky\s(?P<query>.*)", text, re.I|re.U)
+    match = re.match(r"^(?P<mode>[.!@])l(ucky)\s(?P<query>.*)", text, re.I|re.U)
```
Should be `r"^(?P<mode>[.!@])l(ucky)?\s(?P<query>.*)"`

Wessie commented on hanyuu_commands.py:
```diff
@@ -421,7 +421,7 @@ def lucky(server, nick, channel, text, hostmask):
         server.notice(nick, message)
         
-lucky.handler = ("on_text", r'[.!@]lucky\b', irc.ALL_NICKS, irc.MAIN_CHANNELS)
+lucky.handler = ("on_text", r'[.!@]l(ucky)\b', irc.ALL_NICKS, irc.MAIN_CHANNELS)
```
Should be `r'[.!@]l(ucky)?\b'`

Wessie commented on hanyuu_commands.py:
```diff
@@ -690,7 +690,7 @@ def lastfm_setuser(server, nick, channel, text, hostmask):
 
 def favorite_list(server, nick, channel, text, hostmask):
-    match = re.match(r'^(?P<mode>[.!@])fl(ist)?($|\s)(?P<fnick>.*)', text, re.I|re.U)
+    match = re.match(r'^(?P<mode>[.!@])f(ave|avorite)l(ist)?($|\s)(?P<fnick>.*)', text, re.I|re.U)
```
And again, should be `r'^(?P<mode>[.!@])f(ave|avorite)?l(ist)?($|\s)(?P<fnick>.*)'`

Wessie commented on hanyuu_commands.py:
```diff
@@ -707,7 +707,7 @@ def favorite_list(server, nick, channel, text, hostmask):
         server.notice(nick, message)
 
-favorite_list.handler = ('on_text', r'[.!@]fl(ist)?',
+favorite_list.handler = ('on_text', r'[.!@]f(ave|avorite)l(ist)?',
```
`r'[.!@]f(ave|avorite)?l(ist)?'`

Annih7 committed:
Update hanyuu_commands.py

change token comparisons to use multiple comparisons linked with or

add zero-or-one quantifier thing mentioned, hopefully correctly.

Annih7 committed:
Update hanyuu_commands.py

that should do it (regex fixes, just adding some "?"s)

Annih7 committed:
Update hanyuu_commands.py

.random now also works with .ra

Wessie commented on hanyuu_commands.py:
```diff
@@ -362,12 +362,12 @@ def request_from_list(songs):
-    elif (re.match(r"^fave (.*)", command)):
-        fave_nick = re.match(r"^fave (.*)", command).groups()[1]
+    elif (re.match(r"^f(ave|avorite)? (.*)", command)):
+        fave_nick = re.match(r"^f(ave|avorite)? (.*)", command).groups()[1]
```
This captures the wrong group now, since `(ave|avorite)` is now the first group, either make it a non-capturing group or change the index to `2`

Annih7 committed:
Update hanyuu_commands.py

Attempt fix of line 370: change index to 2
